sdk/src/driftClient.ts

```diff
@@ -5239,6 +5245,169 @@ export class DriftClient {
 		);
 	}
 
+	public async signTakerOrderParams(
+		orderParamsMessage: SwiftOrderParamsMessage
+	): Promise<Uint8Array> {
+		const takerOrderParamsMessage =
+			this.getEncodedSwiftOrderParamsMessage(orderParamsMessage);
+		return await this.signMessage(takerOrderParamsMessage);
+	}
+
+	public getEncodedSwiftOrderParamsMessage(
+		orderParamsMessage: SwiftOrderParamsMessage
+	): Uint8Array {
+		return Uint8Array.from(
+			this.program.coder.types.encode(
+				'SwiftOrderParamsMessage',
+				orderParamsMessage
+			)
+		);
+	}
+
+	public async signMessage(message: Uint8Array): Promise<Uint8Array> {
+		return await ed.sign(message, this.wallet.payer.secretKey.slice(0, 32));
+	}
+
+	public assembleSwiftServerMessage(
+		message: Uint8Array,
+		signature: Uint8Array,
+		takerPubkey: PublicKey,
+		marketIndex: number,
+		marketType: MarketType
+	): {
+		message: string;
+		signature: string;
+		taker_pubkey: string;
+		market_index: number;
+		market_type: 'perp' | 'spot';
+	} {
+		return {
+			message: Buffer.from(message).toString('base64'),
+			signature: Buffer.from(signature).toString('base64'),
+			taker_pubkey: takerPubkey.toBase58(),
+			market_index: marketIndex,
+			market_type: isVariant(marketType, 'perp') ? 'perp' : 'spot',
+		};
+	}
+
+	public async placeAndMakeSwiftPerpOrder(
+		takerOrderParamsMessage: SwiftOrderParamsMessage,
+		takerSignature: Uint8Array,
+		takerInfo: {
+			taker: PublicKey;
+			takerStats: PublicKey;
+			takerUserAccount: UserAccount;
+		},
+		orderParams: OptionalOrderParams,
+		referrerInfo?: ReferrerInfo,
+		txParams?: TxParams,
+		subAccountId?: number
+	): Promise<TransactionSignature> {
+		const ixs = await this.getPlaceAndMakeSwiftPerpOrderIxs(
+			takerOrderParamsMessage,
+			takerSignature,
+			takerInfo,
+			orderParams,
+			referrerInfo,
+			subAccountId
+		);
+		const { txSig, slot } = await this.sendTransaction(
+			await this.buildTransaction(ixs, txParams),
+			[],
+			this.opts
+		);
+
+		this.perpMarketLastSlotCache.set(orderParams.marketIndex, slot);
+		return txSig;
+	}
+
+	public async getPlaceAndMakeSwiftPerpOrderIxs(
```

Comment:
people will prob use jit proxy so we should expose method to just get the placeSwiftTakerOrder ix